### PR TITLE
Training loop

### PR DESCRIPTION
**Pull request by Toalaah**

### CODE REVIEW

**Comment by EnnioEnnio** [comment]
lgtm. (even tho I have to admit that I still have some question marks in my head regarding the data loading in general. How to make sure we pass the correct labels together with the images?)

Maybe use tqdm library to track training progress? (see [here](https://tqdm.github.io) - it's already in requirements)

**Comment by Toalaah** [comment]
This is still very much a WIP situation, would not merge this in its current state. Perhaps I shouldn't have created the MR in the first place, but alas. As you mentioned, data loading is still an issue. In fact, if you look carefully you can see that we are not even using any real data at all from the `make_training_set` fn, let alone running the trainer in the `main` loop! Haven't yet gotten around to this (data loading) quite yet, hope to in the next couple of days / weekend.

Of course, if you already have something working feel free to add some commits to this branch, don't mind at all.

**Comment by EnnioEnnio** [comment]
I also start working on that branch now - maybe I'll find something out :)

**Comment by EnnioEnnio** [comment]
Okay, I continued working on that Issue since I couldn't sleep ... 
It's almost done, there is just one minor TODO left (see comment in code) but nothing is tested yet so it might fall apart. 

I tried to include logging using Weights & Biases and sent you an invite via mail. 
I guess it's ready for the first training really soon. 

Since I added a lot feel free to ask whenever anything is unclear - I tried my best commenting everything :) 
Enjoy your weekend 🌞

**Comment by EnnioEnnio** [comment]
Regarding dc57bd27c3e0e37732165abc76241a65be43633d 
We had some mixup of data and training before but now that everything is nicely separated, I feel this should happen on a new branch since it's more related to the data preprocessing and not really relevant for the training loop in its current status, what do you think? 

Also take care about the `make_training_and_validation_set` method as it is currently already able to deal with the subfolder structure via `get_files_from_subfolders `.

**Comment by Toalaah** [comment]
This could be done, sure. I think I added the fn cause I was having troubles with the mentioned function `make_training_and_validation_set`, maybe I was using an already pre-processed folder structure which the function didn't like (although, it just walks through all dirs, so this couldn't have been the issue). either way, just decided to hack this together to quickly get the training loop to run locally.

If you wish, I will rebase the branch and drop the commit, but TBH I don't really think it is necessary as we aren't squashing or anything like that.

**Comment by EnnioEnnio** [comment]
I would suggest just merging into main if it works. We can do some refactoring using the method you mentioned for another iteration, no?

**Comment by EnnioEnnio** [comment]
Just a quick add regarding the function `get_files_from_subfolders`: 
I tested it locally and it gave me a list of all files in the respective category (masked/unmasked).
The Idea was to then pass this lists to `make_training_and_validation_set` which then picks randomly paths from each list, converts them into image objects assigning the corresponding label putting it in the training or validation set and returning it. 
I just tested the path extraction but not the rest so this might have been the issue - not sure tho
